ext/php8/configuration.c

```diff
@@ -30,8 +30,24 @@ void ddtrace_reload_config(void) {
         free(ddtrace_memoized_configuration.getter_name); \
     }                                                     \
     ddtrace_memoized_configuration.__is_set_##getter_name = FALSE;
-#define HASH(getter_name, ...)                                     \
-    zend_hash_release(ddtrace_memoized_configuration.getter_name); \
+// this will leak memory if RC != 1, but prevents crashes. This could only ever happens if an user uses
+// \dd_trace_internal_fn('ddtrace_reload_config'); at the wrong point in time - which is an internal testsuite function
+// which users are not supposed to use
+#define HASH(getter_name, ...)                                                                    \
```

Comment:
ZAI config is supposed to improve upon this... Don't give this too much scrutiny, it works...